jit-compiler/src/codegen.rs

```diff
@@ -125,7 +148,7 @@ impl<'a, T: FieldElement> CodeGenerator<'a, T> {
         expr: &Expression,
```

Comment:
can't this function receive the `LambdaExpression` directly to avoid the `match` and `panic`?